From fd9a12f9fdd9dd3e66c59dd7037e864b948085f7 Mon Sep 17 00:00:00 2001
From: Harish Mahendrakar <harish.mahendrakar@ittiam.com>
Date: Thu, 15 Dec 2016 10:55:28 +0530
Subject: Decoder: Fixed cur_mb_info initialization in error cases

When number of MBs decoded in the current frame is zero and an error is
detected, MB info was unnecessarily decremented in multi-core case.

Bug: 33351708
Bug: 33298089
Bug: 33250932
Change-Id: I150c48c6fb382736f1de855ba5384c12bf84710d
(cherry picked from commit 902c15251fe53f103f910b9f6a133e4cc7ca7416)
---
 decoder/ih264d_parse_pslice.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/decoder/ih264d_parse_pslice.c b/decoder/ih264d_parse_pslice.c
index e2acd4a..63fa179 100644
--- a/decoder/ih264d_parse_pslice.c
+++ b/decoder/ih264d_parse_pslice.c
@@ -1596,7 +1596,7 @@ WORD32 ih264d_mark_err_slice_skip(dec_struct_t * ps_dec,
             {
                 if(ps_dec->u1_separate_parse)
                 {
-                    ps_cur_mb_info = ps_dec->ps_nmb_info - 1;
+                    ps_cur_mb_info = ps_dec->ps_nmb_info;
                 }
                 else
                 {
-- 
cgit v1.1

